File: src/sshSegment.js
import {fg} from './segments'

const bg0 = '#843200'
const fg0 = '#D35400'
const fg1 = '#ffffff'

const clientIp = (client) => client.replace(/((?:\d+\.)+(?:\d+)).+/, '$1')

export default (tty, client) => {
  return (!tty && !client) ? {
    text: null
  } : {
    bg0, fg0,
    text: `\uf1be  ${fg(fg1)}${tty} ${clientIp(client)}`
  }
}

File: src/segments/sshSegment.rs
#![allow(non_upper_case_globals)]
#![allow(dead_code)]

use ::segments::*;
use std::env;

const bg0: u32 = 0xd35400;
const fg0: u32 = 0x094d77;
const fg1: u32 = 0xbee3f7;

pub fn segment() -> Option<Vec<Part>> {
  if env::var("SSH_TTY").ok().map_or(0, |tty| tty.len()) > 0 {
    let user = env::var("USER").ok().unwrap_or_else(|| "".to_string());
    let host = env::var("HOST").ok().unwrap_or_else(|| "".to_string());
    let result = vec![
      Part::Bg(bg0), Part::Fg(fg0), Part::StaticText("\u{f0c2}  "), 
      Part::Fg(fg1), Part::Text(user), Part::StaticText("@"), Part::Text(host)
    ];
    Some(result)
  }
  else {
    None
  }
}


